src/cryptonote_core/asset_history_utils.cpp

```
@@ -0,0 +1,592 @@
+
+    std::string op_reason;
+    if (!validate_asset_descriptor_operation(op, op_reason))
+    {
+      reason = "invalid asset descriptor operation: " + op_reason;
+      return false;
+    }
+
+    // ── Mandatory fan-out: deploy and emit must create >= MIN_ASSET_EMISSION_OUTPUTS
+    // tx_out_zarcanum outputs so that ring members exist from the first block.
+    // The wallet auto-generates self-sends to reach this minimum.
+    if (hf_version >= feature::CONFIDENTIAL_ASSETS)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)